### ISSUE TITLE
Moodle 2.9 breaks plugin-install

### ISSUE THREAD

**Issue by usernamenumber** [tmuras/moosh#101]
It looks like 2.9 no longer has a `lib/pluginlib.php`, which seems to have been moved to [`lib/classes/plugin_manager.php`](https://github.com/moodle/moodle/blob/MOODLE_29_STABLE/lib/classes/plugin_manager.php)

Obviously this breaks the `plugin-install` command, and probably others. :(

**Comment by usernamenumber** [comment]
Is there an ETA on this? (I know, I know, "patches welcome" ;)  I appreciate the volunteer effort behind this tool, I'd just like to know whether I should not expect to be able to use it with my 2.9 site for a while?

(btw, if @tmuras or anyone else involved with moosh will be at the MoodleMoot in MN next month, I'd be down for a hackathon if we can find the time)

**Comment by tmuras** [comment]
Hi Brad, I have a lot to catch up and no free time unfortunately, so I'd say don't use it with 2.9 for now.

**Comment by dkns** [comment]
@usernamenumber could you post example command that you're using and error message that you get?

**Comment by usernamenumber** [comment]
Here are a couple of examples:

```
$ moosh plugin-install enrol_autoenrol 29
PHP Warning:  require_once(/www/training.nocc/moodle29/moodle/lib/pluginlib.php): failed to open stream: No such file or directory in /usr/share/moosh/Moosh/Command/Generic/Plugin/PluginInstall.php on line 30

Warning: require_once(/www/training.nocc/moodle29/moodle/lib/pluginlib.php): failed to open stream: No such file or directory in /usr/share/moosh/Moosh/Command/Generic/Plugin/PluginInstall.php on line 30
PHP Fatal error:  require_once(): Failed opening required '/www/training.nocc/moodle29/moodle/lib/pluginlib.php' (include_path='/www/training.nocc/moodle29/moodle/lib/zend:/www/training.nocc/moodle29/moodle/lib/pear:.:/usr/share/php:/usr/share/pear') in /usr/share/moosh/Moosh/Command/Generic/Plugin/PluginInstall.php on line 30

Fatal error: require_once(): Failed opening required '/www/training.nocc/moodle29/moodle/lib/pluginlib.php' (include_path='/www/training.nocc/moodle29/moodle/lib/zend:/www/training.nocc/moodle29/moodle/lib/pear:.:/usr/share/php:/usr/share/pear') in /usr/share/moosh/Moosh/Command/Generic/Plugin/PluginInstall.php on line 30
```

```
$ moosh plugin-list enrol

PHP Notice:  Trying to get property of non-object in /usr/share/moosh/Moosh/Command/Generic/Plugin/PluginList.php on line 50

Notice: Trying to get property of non-object in /usr/share/moosh/Moosh/Command/Generic/Plugin/PluginList.php on line 50
PHP Notice:  Trying to get property of non-object in /usr/share/moosh/Moosh/Command/Generic/Plugin/PluginList.php on line 50

Notice: Trying to get property of non-object in /usr/share/moosh/Moosh/Command/Generic/Plugin/PluginList.php on line 50
```

**Comment by dkns** [comment]
@usernamenumber I'm guessing you've installed moosh using .deb package? plugin-list doesn't accept any arguments at current commit. 

Regarding plugin-install it should be fixed for 2.9 (but I might've broken it for older versions of moodle in the process, ooops). If you're feeling adventurous you could pull newest version from git, test it a bit and let me know if there are more issue with this command.

**Comment by usernamenumber** [comment]
I think by the time I got this bug I'd already switched to the git version,
but the apt version is also installed, so perhaps there's some confusion.
The next time I'm at my computer I'll uninstall the apt version, pull the
largest from git, and test. Thanks for looking into it!
